eclipse-scout-core/test/lookup/RestLookupCallSpec.js

```diff
@@ -0,0 +1,327 @@
+
+    const expectedDefaultRestriction = {
+      active: true,
+      maxRowCount: 777
+    };
+
+    expect(lookupCall.restriction).toBeNull();
+    expect(lookupCall._restriction).toBeNull();
+
+    const cloneAll = lookupCall.cloneForAll();
+    expect(cloneAll.restriction).toBeNull();
+    expect(cloneAll._restriction).toEqual(expectedDefaultRestriction);
+    cloneAll.addRestriction('myProperty', 'someValue');
+    cloneAll.execute();
+    expect(cloneAll._getRestrictionForAjaxCall()).toEqual({
+      ...expectedDefaultRestriction,
+      myProperty: 'someValue'
+    });
+
+    const cloneText = lookupCall.cloneForText('xyz');
+    expect(cloneText.restriction).toBeNull();
+    expect(cloneText._restriction).toEqual(expectedDefaultRestriction);
+    expect(cloneText.searchText).toBe('xyz'); // not (yet) on the restriction!
+    cloneText.addRestriction('myProperty', 'someValue');
+    cloneText.execute();
+    expect(cloneText._getRestrictionForAjaxCall()).toEqual({
+      ...expectedDefaultRestriction,
+      text: 'xyz',
+      myProperty: 'someValue'
+    });
+
+    const cloneRec = lookupCall.cloneForRec(789);
+    expect(cloneRec.restriction).toBeNull();
+    expect(cloneRec._restriction).toEqual(expectedDefaultRestriction);
+    expect(cloneRec.parentKey).toBe(789); // not (yet) on the restriction!
+    cloneRec.addRestriction('myProperty', 'someValue');
+    // TODO uncomment the lines marked with '<--' when getByRec() is implemented for RestLookupCall.js!
+    // cloneRec.execute(); <-- uncomment
+    expect(cloneRec._getRestrictionForAjaxCall()).toEqual({
+      ...expectedDefaultRestriction,
+      // parentKey: 789, <-- uncomment
+      myProperty: 'someValue'
+    });
+
+    const cloneKey = lookupCall.cloneForKey(123);
+    expect(cloneKey.restriction).toBeNull();
+    expect(cloneKey._restriction).toBeNull(); // different from all, text and rec
+    expect(cloneKey.key).toBe(123); // not (yet) on the restriction!
+    cloneKey.execute();
+    cloneKey.addRestriction('myProperty', 'someValue');
+    expect(cloneKey._getRestrictionForAjaxCall()).toEqual({
+      ids: [123],
+      myProperty: 'someValue'
+    });
+
+    const cloneKeys = lookupCall.cloneForKeys([123, 456]);
+    expect(cloneKeys.restriction).toBeNull();
+    expect(cloneKeys._restriction).toBeNull(); // different from all, text and rec
+    expect(cloneKeys.keys).toEqual([123, 456]); // not (yet) on the restriction!
+    cloneKeys.addRestriction('myProperty', 'someValue');
+    cloneKeys.execute();
+    expect(cloneKeys._getRestrictionForAjaxCall()).toEqual({
+      ids: [123, 456],
+      myProperty: 'someValue'
+    });
+
+    // Check template again (must not have been altered)
+    expect(lookupCall.restriction).toBeNull();
+    expect(lookupCall._restriction).toBeNull();
+  });
+
+  it('applies restrictions in the expected order', () => {
+    // This test asserts that the following precedence rules are applies then creating the final restriction object:
```

Comment:
The list does not match the one in the lookup call comment.

Reply:
Correct, I updated the comment